src/Tests/dotnet-new.Tests/TemplateEngineSamplesTest.cs

```diff
@@ -0,0 +1,98 @@
+        [InlineData("06-console-csharp-fsharp", "sample06", new[] { "--language", "F#" }, "multiple languages supported. This one creates F# language template")]
+        [InlineData("07-param-with-custom-short-name", "sample07", null, "customised parameter name")]
+        [InlineData("09-replace-onlyif-after", "sample09", new[] { "--backgroundColor", "grey" }, "replacing with onlyif condition")]
+        [InlineData("10-symbol-from-date", "sample10", null, "usage of date generator")]
+        [InlineData("11-change-string-casing", "sample11", null, "usage of casing generator")]
+        [InlineData("13-constant-value", "sample13", null, "replacing of constant value")]
+        [InlineData("15-computed-symbol", "sample15", null, "usage computed symbols")]
+        [InlineData("16-string-value-transform", "sample16", null, "usage of derived parameter")]
+        public async void TemplateEngineSamplesProjectTest(
+            string folderName,
+            string shortName,
+            string[] arguments,
+            string caseDescription)
+        {
+            _log.LogInformation($"Template with {caseDescription}");
+            Dictionary<string, string> environmentUnderTest = new() { ["DOTNET_NOLOGO"] = false.ToString() };
+            TestContext.Current.AddTestEnvironmentVariables(environmentUnderTest);
+
+            TemplateVerifierOptions options = new TemplateVerifierOptions(templateName: shortName)
+            {
+                TemplateSpecificArgs = arguments ?? Enumerable.Empty<string>(),
+                VerifyCommandOutput = true,
+                SnapshotsDirectory = "Approvals",
+                SettingsDirectory = _sharedHome.HomeDirectory,
+                DoNotAppendTemplateArgsToScenarioName = true,
+                DotnetExecutablePath = TestContext.Current.ToolsetUnderTest.DotNetHostPath,
+                DoNotPrependCallerMethodNameToScenarioName = true,
+                ScenarioName = $"{folderName.Substring(folderName.IndexOf("-") + 1)}{GetScenarioName(arguments)}"
+            }
+            .WithCustomEnvironment(environmentUnderTest)
+            .WithCustomScrubbers(
+               ScrubbersDefinition.Empty
+               .AddScrubber(sb => sb.Replace(DateTime.Now.ToString("MM/dd/yyyy"), "**/**/****")));
+
+            VerificationEngine engine = new VerificationEngine(_log);
+            await engine.Execute(options)
+                .ConfigureAwait(false);
+        }
+
+        private string GetScenarioName(string[]? args)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            if (args != null)
+            {
+                sb.Append('.');
+
+                for (int index = 0; index < args.Length; index += 2)
+                {
+                    sb.Append($"{args[index].Replace("--", "")}={args[index + 1]}");
+                    if (index < args.Length - 2)
+                    {
+                        sb.Append('.');
+                    }
```

Comment:
This is completely fine as is. 

Just a super optional (and opinionated) suggestion on shortening the code - the string can be composed unconditionally with the separating dot (`sb.Append($"{args[index].Replace("--", "")}={args[index + 1]}.");`) and then just in the final `ToString` we can slash the last dot (`sb.ToString(0, Math.Max(0,sb.Length-2))`)